src/fabric.c

```diff
@@ -485,10 +498,45 @@ void ofi_create_filter(struct fi_filter *filter, const char *raw_filter)
 		++raw_filter;
 	}
 
-	filter->names= ofi_split_and_alloc(raw_filter, ",", NULL);
+	filter->names = ofi_split_and_alloc(raw_filter, ",", NULL);
 	if (!filter->names)
 		FI_WARN(&core_prov, FI_LOG_CORE,
 			"unable to parse filter from: %s\n", raw_filter);
+
+	tkn_str = strtok((char *)raw_filter, ";");
+	if (!tkn_str)
+		return;
+
+	while (tkn_str != NULL) {
+		count_prov_names++;
+		tkn_prove_names[i++] = tkn_str;
+		tkn_str = strtok (NULL, ";");
+	}
```

Comment:
We can drop the above code.  filter->names contains an array of entries in the filter, with the last entry being null.

Reply:
@shefty Done with a note*. 
Array `filter->names`, which contains an array of entries in the filter, contains single and , unfortunately, multi  "full string". 
E.g. when we give `"verbs;ofi_rxd"` to `FI_PROVIDER`, the output of `filter->names[i]` will give us a full string `"verbs;ofi_rxd"`. it is because  of `ofi_split_and_alloc(r...",", ...);` - `,` .

*- To resolve this, I bring `char **filter_strs_to_parse;`  in order to parse gotten filter provider names by ";" using `ofi_split_and_alloc(...)`